text/0000-gatsby-plugin-cli.md

```diff
@@ -0,0 +1,112 @@
+Thanks everyone for checking this out. I've started on the code and am a happy to be the one to implement most/all of this. Thanks for the feedback all keep up the great work on gatsby.
+
+
+- Start Date: 2018-09-28
+- RFC PR: (leave this empty)
+- Gatsby Issue: (leave this empty)
+
+# Summary
+
+I find it  tedious to have to do the same 3 steps every time I add a plugin to gatsby.
+
+* `npm instal *plugin*`
+* Check plugin docs for correct -`gatsby-config.js` syntax
+* modify `gatsby-config.js`
+
+I'm suggesting we create an additional command to add and remove plugins using the `gatsby-cli`.
+
+# Basic example
+
+A new work flow for installing a plugin could look something like this:
+
+* `gatsby plugin add gatsby-transformer-x` (this would run npm install for the plugin and inject an initial config into `gatsby-config.js`)
+
+* For plugins that don't require additional config we're done. If they do I at least have the basics that I can probably understand how to modify from a quick glance. ie. `path: /your/path/to/xyz.whatever` I know what this needs to point to and I can do that without checking docs.
+
+* if needed then the dev can check docs.
+
+
+A new work flow for removing a plugin could look like:
+ 
+ * `gatsby plugin remove gatsby-transformer-x` (This would just remove the config for that plugin and remove the package using npm from the `package.json` and node_modules.)
+
+# Motivation
+
+The problem I'm trying to solve is simplifying and accelerating the gatsby work flow in relation to managing plugins over the course of development.
+
+The major two use cases I can think to support would be installation and un-installation of plugins. Though I could see a potential use for disabling plugins (though commenting out code in the config isn't too difficult) and searching for plugins.
+
+# Detailed design
+
+## CLI Changes
+
+`plugin [action] [plugin-name]`
+
+### Actions
+
+* add\* - Installs npm package, adds plugin to `gatsby-config.js`, warns if further config is needed. potential alias: install
+* remove\* - Removes npm package form `node_modules` and `package.json` and removes config from `gatsby-config.js`. potential alias: uninstall
+* disable - comments out config in `gatsby-config.js`.
+* enable - un-comments config in `gatsby-config.js`.
```

Comment:
This should be possible to handle. It would be a matter of checking the `package.json` before doing the `npm install` and checking the `gatsby-config.js` before writing to it.

Reply:
Oh I get your question, I don't think I understood it at first, yeah, parsing comments could be difficult. I didn't love the enable/disable options either, so we'll just forget about them.